Feat(web): Extend Tooltip Floating UI example, deprecate CSS modifier… (#1164)
lmc-eu/spirit-design-system: packages/web/src/scss/components/Tooltip/README.md

Bare Tooltip HTML:

```html
<div class="Tooltip" data-spirit-placement="top">
  Hello there!
  <span class="Tooltip__arrow"></span>
</div>
```

## Feature Flag: Data Selector Placement

> Tooltip placement is currently set using CSS modifiers. In the future it will be set using `data-spirit-placement`

Reviewer: ```suggestion
Tooltip placement is currently using CSS modifiers. In the future it will be using `data-spirit-placement`
```

?

</div>
```

## Feature Flag: Data Selector Placement

Tooltip placement is currently set using CSS modifiers. In the future it will be set using `data-spirit-placement`
attribute in order to make the placement independent of the component and compatible with Floating UI.

> To enable this behavior now you can enable the feature flag, either set the `$tooltip-enable-data-placement`

Reviewer 2: ```suggestion
To enable this behavior now you can use the feature flag, either set the `$tooltip-enable-data-placement`
```

## Placement

Tooltip implements the [Placement Dictionary][dictionary-placement] for placement. The dictionary values are used as
a value of data attribute `data-spirit-placement`, e.g. `data-spirit-placement="top"`, `data-spirit-placement="rightEnd"`, etc.

> For JS-controlled positioning please use the `data-spirit-tooltip-controlled` attribute instead of

Reviewer 2: ```suggestion
For JS-controlled positioning please use the `data-spirit-placement-controlled` attribute instead of
```

As we have discussed personally. Or use some CSS class instead ;-) Thanks.

## Placement

Tooltip implements the [Placement Dictionary][dictionary-placement] for placement. The dictionary values are used as

> a value of data attribute `data-spirit-placement`, e.g. `data-spirit-placement="top"`, `data-spirit-placement="rightEnd"`, etc.

Reviewer: ```suggestion
a value of data attribute `data-spirit-placement`, e.g. `data-spirit-placement="top"`, `data-spirit-placement="right-end"`, etc.
```

> For JS-controlled positioning please use the `data-spirit-placement-controlled` attribute instead of

Reviewer: ```suggestion
For JS-controlled positioning please add the `data-spirit-placement-controlled` attribute instead of specifying the placement using the
```

I'm not sure if it's clear that adding just the attribute itself is fine.

`data-spirit-placement` modifiers (more on that [below](#advanced-positioning)).

```html

> <div class="Tooltip" data-spirit-placement="rightStart">

Reviewer: ```suggestion
<div class="Tooltip" data-spirit-placement="right-start">
```

</div>
```

> If you enabled the [Data Selector Placement feature flag](#feature-flag-data-selector-placement),

Reviewer: ```suggestion
If you have the [Data Selector Placement feature flag](#feature-flag-data-selector-placement) enabled,
```

or

```suggestion
If you have enabled the [Data Selector Placement feature flag](#feature-flag-data-selector-placement),
```

(who knows)

set `data-spirit-placement-controlled` on the `.Tooltip` to control it and prevent conflicts with
the default CSS positioning.

```html
<div id="my-advanced-tooltip" class="Tooltip" data-spirit-placement-controlled>
  Hello there!
  <span class="Tooltip__arrow"></span>
</div>
```

### Arrow

When controlling Tooltip arrow with JavaScript, set `data-spirit-element="arrow"`
on the `.Tooltip__arrow` to control it and prevent conflicts with the default CSS positioning.

```html
<div id="my-advanced-tooltip" class="Tooltip" data-spirit-placement-controlled>
  Hello there!
  <span class="Tooltip__arrow" data-spirit-element="arrow"></span>
</div>
```

### Example

> 💻 Check our minimum [example] that uses external library

Reviewer: 🎉
